components/ILIAS/Forum/classes/Thread/ForumThreadTableSessionStorage.php

```diff
@@ -0,0 +1,176 @@
+    {
+        $sortation = $this->getThreadSortation();
+        $page = $this->getThreadPage();
+        $limit = ilForumProperties::PAGE_SIZE_THREAD_OVERVIEW;
+
+        $params = [
+            'is_moderator' => $this->is_moderator,
+            'excluded_ids' => [],
+            'order_column' => $sortation->field(),
+            'order_direction' => $sortation->direction()
+        ];
+
+        $data = $forum->getAllThreads(
+            $topicData->getTopPk(),
+            $params,
+            $limit,
+            $page * $limit
+        );
+        if ($data['items'] === [] && $page > 0) {
+            ilSession::set($this->buildSessionKey($this->forum_ref_id, self::KEY_THREAD_PAGE), 0);
+            $data = $forum->getAllThreads(
+                $topicData->getTopPk(),
+                $params,
+                $limit,
+                $page * $limit
+            );
+        }
+
+        $temp_data = (array_map(static function (ilForumTopic $thread): array {
+            return ['thread' => $thread];
+        }, $data['items']));
+
+        $thread_ids = [];
+        $user_ids = [];
+        foreach ($data['items'] as $thread) {
+            /** @var ilForumTopic $thread */
+            $thread_ids[] = $thread->getId();
+            if ($thread->getDisplayUserId() > 0) {
+                $user_ids[$thread->getDisplayUserId()] = $thread->getDisplayUserId();
+            }
+        }
+
+        $user_ids = array_merge(
+            ilObjForum::getUserIdsOfLastPostsByRefIdAndThreadIds($this->forum_ref_id, $thread_ids),
+            $user_ids
+        );
+
+        ilForumAuthorInformationCache::preloadUserObjects(array_unique($user_ids));
+
+        return $temp_data;
+    }
+
+    public function getThreadSortation(): ThreadSortation
+    {
+        $query_thread_sortation = $this->getKeyValueFromQuery(self::KEY_THREAD_SORTATION, ThreadSortation::DEFAULT_SORTATION->value);
+
+        if ($query_thread_sortation !== ThreadSortation::DEFAULT_SORTATION->value) {
+            $this->setSessionKeyValue($this->forum_ref_id, self::KEY_THREAD_SORTATION, $query_thread_sortation);
+            return ThreadSortation::tryFrom(
+                $query_thread_sortation
+            );
+        }
+
+        $session_thread_sortation = $this->getKeyValueFromSession(
+            $this->forum_ref_id,
+            self::KEY_THREAD_SORTATION,
+            ThreadSortation::DEFAULT_SORTATION->value
+        );
+
+        return ThreadSortation::tryFrom(
+            $session_thread_sortation
+        );
+    }
+
+    public function getThreadPage(): int
+    {
+        $query_thread_page = $this->getKeyValueFromQuery(self::KEY_THREAD_PAGE, 0);
+
+        if ($query_thread_page !== 0) {
+            $this->setSessionKeyValue($this->forum_ref_id, self::KEY_THREAD_PAGE, $query_thread_page);
+            return $query_thread_page;
+        }
+
+        return $this->getKeyValueFromSession(
```

Comment:
I might be wrong, but if the queried page is `0`, why do you read the page from the session storage? `0` translates to "First Page", which is valid. I think we need a different default value when executing `$this->getKeyValueFromQuery(self::KEY_THREAD_PAGE, 0);`, e.g. `null` or `-1`.